### PR TITLE
Update localizations

### LINKED ISSUE #65018: DateRangePicker has wrong input hint in french
Issue to track fix made in https://github.com/flutter/flutter/pull/64867.

Instead of `'mm/jj/aaaa'`, `dateHelpText` should be `'jj/mm/aaaa'` in `fr`. I'll make a translation request in the google translation console to get this modified.

cc/ @fodil-a

### PR DESCRIPTION
**Pull request by shihaohong**
## Description

Updated localizations based on latest translations from the Google team.

~~I also noticed that the translated text for `searchFieldLabel` appears to have "regressed", but was overridden to be in English a day after the localized versions were generated on the translation console. I'll need to make a request to see if this was intentional or not. I may opt to manually remove these changes in the meantime if that makes more sense.~~ <-- These were not intended

## Related Issues

Fixes https://github.com/flutter/flutter/issues/65018
Fixes part of https://github.com/flutter/flutter/issues/63372, the originally requested locales of `af`, `fr`, `hy`, `nl`, `cs`, `sk` and `hu`.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by shihaohong** [comment]
Edit: I'll wait until the translation team comes back with an answer regarding the "searchFieldLabel" text before moving forward with this update.

**Review by HansMuller — APPROVED** [review]
LGTM modulo the pending update (or not) to the searchFieldLabel strings.

**Pull request merged**
